app/controllers/responses_controller.rb

```diff
@@ -112,33 +115,34 @@ def destroy
   end
 
   def possible_submitters
-    # get the users to which this response can be assigned
-    # which is the users in this mission plus the submitter of this response
-    @possible_submitters = User.assigned_to_or_submitter(current_mission, @response).by_name
-    @possible_submitters = apply_search_if_given(User, @possible_submitters)
-    @possible_submitters = @possible_submitters.paginate(page: params[:page], per_page: 20)
-
-    render(json: {possible_submitters: ActiveModel::ArraySerializer.new(@possible_submitters),
-                  more: @possible_submitters.next_page.present?}, select2: true)
+    users = if params[:response_id].present?
+              response = Response.find(params[:response_id])
+              response_user_id = response.try(:user_id)
+              User.assigned_to(current_mission).or(User.where(id: response_user_id)).by_name
+            else
+              User.assigned_to(current_mission).by_name
+            end
+    render_possible_users(users)
   end
 
-  def possible_users
-    search_mode = params[:search_mode] || "submitters"
-
-    @possible_users =
-      case search_mode
-      when "submitters" then User.assigned_to_or_submitter(current_mission, @response).by_name
-      when "reviewers" then User.with_roles(current_mission, %w[coordinator staffer reviewer]).by_name
-      end
-    @possible_users = apply_search_if_given(User, @possible_users)
-    @possible_users = @possible_users.paginate(page: params[:page], per_page: 20)
-
-    render(json: {possible_users: ActiveModel::ArraySerializer.new(@possible_users),
-                  more: @possible_users.next_page.present?}, select2: true)
+  def possible_reviewers
+    users = User.with_roles(current_mission, %w[coordinator staffer reviewer]).by_name
+    render_possible_users(users)
   end
 
   private
 
+  def render_possible_users(possible_users)
+    @possible_users = apply_search_if_given(User, possible_users)
+      .natural_sort_by_key
+      .paginate(page: params[:page], per_page: 20)
```

Comment:
`natural_sort` is needed before `paginate`, but that wrecks the nice lazy-evaluation for ActiveRecord Relations. @smoyth do you think this will significantly impact performance? Any simple way to load-test this?